TRANSLATING.md

```diff
@@ -0,0 +1,161 @@
+| `manifests`              | `script-lab-prod.xml`     | Manifest for the production version.           |
+
+<a id="create"></a>
+## Create a new translation
+
+When creating a new translation, some steps must be done before you can start to translate.
+
+The first step is to retrieve the official code for your language. You can refer to this list [Language identifiers and OptionState Id values in Office 2016](https://technet.microsoft.com/en-us/library/cc179219(v=office.16).aspx). You need the first two letters of the mentionned codes. For example, for German the code is `de`, for French `fr` and for Spanish `es`.
+
+### Server part
+
+Create a copy of the file `english.ts` from and to the folder `src/server/strings` and rename this copy to the language you are going to translate the strings. Please use the English name for your language. For example, if you are going to translate the strings into French, rename the file to `french.ts`. The filename must be in lower case letters.
+
+Open the new file and change the word `English` of the function name `getEnglishStrings()` to the name of your language, where the first letter must be upper case. If we keep the example of creating a french translation, the function name would now be `getFrenchStrings()`.
+
+Next step would be to tell the code, that a new language is available. For this, open the file `index.ts` in the server part folder `src/server/strings`. You will have to add two new lines to the code. If we keep our example for French again, the result will be similar to the code block shown below:
+
+```ts
+import { getEnglishStrings } from './english';
+import { getGermanStrings } from './german';
+import { getFrenchStrings } from './french';
+
+const languageGenerator: { [key: string]: () => ServerStrings } = {
+    'en': () => getEnglishStrings(),
+    'de': () => getGermanStrings(),
+    'fr': () => getFrenchStrings(),
+    '??': () => createFakeStrings(() => getEnglishStrings())
+};
+```
+In the code above, the new line `import { getFrenchStrings } from './french';` has been added to the already existing imports for English and German. And the second new code line `'fr': () => getFrenchStrings(),` has been added to the language generator.
+
+That's all. Now you can start to translate your language file. Please note, that while most strings are just strings, others are functions that accept some parameters, and those parameters often get inlined into the return string using TypeScript template strings such as double quotes `"`. For these variables, which look like `${parameterName}`, the name inside of `${parameterName}` should be preserved as is.
+
+The strings in the TypeScripe files are enclosed by single quotes or apostrophes `'`. If you need to use the single quote character inside a string, you need to escape this character by prepending a backslash. Example: the word `today` then translates in french to `aujourd\'hui`.
```

Comment:
Actually (both FYI, and something to put in here), the simplest way to avoid the escaping problem is to use the TypeScript template strings, even if you don't do any variables within them!

so use the ` symbol, as so:

    something: `It's a wonderful "thing" that you don't have to escape anything anymore!`